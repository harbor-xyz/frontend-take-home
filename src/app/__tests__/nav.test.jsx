import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import MainNav from '../components/organisms/mainNav';;

describe('MainNav', () => {
  it('renders MainNav unchanged', () => {
    const { container } = render(<MainNav />)
    expect(container).toMatchSnapshot()
  })

  it('renders all menus', () => {
    render(<MainNav />)

    const First = screen.getByText('Projects');
    const Second = screen.getByText('Docs');
    const Third = screen.getByText('Command cheatsheet');
    const Fourth = screen.getByText('Your user key');

    expect(First).toBeInTheDocument()
    expect(Second).toBeInTheDocument()
    expect(Third).toBeInTheDocument()
    expect(Fourth).toBeInTheDocument()
  })
})