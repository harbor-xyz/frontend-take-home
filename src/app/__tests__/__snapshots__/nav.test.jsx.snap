// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav renders MainNav unchanged 1`] = `
<div>
  <nav
    class="mainNavContainer"
  >
    <svg
      class="brand"
      fill="none"
      viewBox="0 0 111 26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#hl-a)"
        fill="#fff"
        opacity=".9"
      >
        <g
          clip-path="url(#hl-b)"
        >
          <path
            d="M25.74 8.353v3.493h-6.628c-1.018 0-2.004.36-2.782 1.014l-4.163 3.5a7.021 7.021 0 0 1-4.507 1.642H.74v-3.495l6.627.003a4.33 4.33 0 0 0 2.782-1.014l4.163-3.5a7.015 7.015 0 0 1 4.505-1.643H25.739ZM25.74 14.51v3.492h-4.67c-.837 0-1.652.297-2.294.837l-5.744 4.83v-4.562l4.008-3.367a5.232 5.232 0 0 1 3.372-1.23h5.327ZM13.446 2.684v4.562l-4.007 3.37a5.242 5.242 0 0 1-3.373 1.23H.74V8.353h4.666c.838 0 1.653-.297 2.295-.836l5.746-4.833Z"
          />
        </g>
        <g
          clip-path="url(#hl-c)"
        >
          <path
            d="M106.108 15.308a4.414 4.414 0 0 0 1.955-3.671v-1.85a4.425 4.425 0 0 0-4.416-4.422h-6.684V20.91h3.242v-4.793h2.71l2.141 4.793h3.456l-2.403-5.603h-.001Zm-1.161-3.942a1.83 1.83 0 0 1-1.828 1.828h-2.915V8.33h2.915c1.008 0 1.828.821 1.828 1.83v1.204ZM91.825 5.03a4.17 4.17 0 0 0-.875-.093h-2.533c-.281 0-.52.026-.82.093a4.195 4.195 0 0 0-3.299 4.086v7.988a4.192 4.192 0 0 0 3.3 4.084c.302.063.54.09.82.09h2.532c.302 0 .597-.03.873-.09a4.193 4.193 0 0 0 3.3-4.084V9.116a4.196 4.196 0 0 0-3.298-4.086Zm.112 11.78c0 .884-.72 1.604-1.602 1.604h-1.25c-.883 0-1.601-.72-1.601-1.605V9.405c0-.882.718-1.602 1.602-1.602h1.249c.883 0 1.602.72 1.602 1.602v7.404ZM81.524 12.888A4.391 4.391 0 0 0 82.81 9.77a4.408 4.408 0 0 0-4.4-4.407h-6.527v15.547h6.525c2.427 0 4.403-1.992 4.403-4.44v-.19c0-1.236-.52-2.42-1.43-3.258.048-.042.096-.087.143-.135Zm-6.626-4.695h3.143a1.759 1.759 0 0 1 1.241 3.002 1.74 1.74 0 0 1-1.241.513h-3.143V8.193Zm3.143 9.774h-3.143v-3.515h3.143a1.758 1.758 0 0 1 0 3.515ZM67.963 15.308a4.413 4.413 0 0 0 1.954-3.671v-1.85a4.422 4.422 0 0 0-4.413-4.422h-6.687V20.91h3.241v-4.793h2.711l2.14 4.793h3.457l-2.403-5.603ZM66.8 11.366a1.83 1.83 0 0 1-1.827 1.828h-2.916V8.33h2.916a1.83 1.83 0 0 1 1.827 1.83v1.204ZM52.624 4.952h-2.692a4.18 4.18 0 0 0-4.17 4.179v11.78h3.347v-4.788h4.337v4.787h3.347V9.132a4.179 4.179 0 0 0-4.17-4.18h.001Zm.822 8.242H49.11V9.598c0-.843.684-1.525 1.523-1.525h1.291c.842 0 1.523.682 1.523 1.525v3.595ZM43.338 5.364v15.544h-3.345v-4.785h-4.341v4.785H32.31V5.364h3.342v7.832h4.341V5.364h3.345Z"
          />
        </g>
      </g>
      <defs>
        <clippath
          id="hl-a"
        >
          <path
            d="M.74.806h110v25.041H.74z"
            fill="#fff"
          />
        </clippath>
        <clippath
          id="hl-b"
        >
          <path
            d="M.74.806h25v25.041h-25z"
            fill="#fff"
          />
        </clippath>
        <clippath
          id="hl-c"
        >
          <path
            d="M31.99 4.562h76.874v17.111H31.99z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
    <a
      class="mainMenuItem undefined"
      href="#"
    >
      <svg
        class="icon"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M40 48c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24V72c0-13.3-10.7-24-24-24H40zm152 16c-17.7 0-32 14.3-32 32s14.3 32 32 32h288c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32h288c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32h288c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zM16 232v48c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24v-48c0-13.3-10.7-24-24-24H40c-13.3 0-24 10.7-24 24zm24 136c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h48c13.3 0 24-10.7 24-24v-48c0-13.3-10.7-24-24-24H40z"
        />
      </svg>
      Projects
    </a>
    <a
      class="mainMenuItem undefined"
      href="#"
    >
      <svg
        class="icon"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96v320c0 53 43 96 96 96h320c17.7 0 32-14.3 32-32s-14.3-32-32-32v-64c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H96zm0 384h256v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48h192c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
        />
      </svg>
      Docs
    </a>
    <a
      class="mainMenuItem undefined"
      href="#"
    >
      <svg
        class="icon"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M41.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 41.4 86.6zM288 416h288c17.7 0 32 14.3 32 32s-14.3 32-32 32H288c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
        />
      </svg>
      Command cheatsheet
    </a>
    <a
      class="mainMenuItem rightSubMenu"
      href="#"
    >
      <svg
        class="icon"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 224c-44.2 0-80-35.8-80-80s35.8-80 80-80 80 35.8 80 80-35.8 80-80 80zm0 64c79.5 0 144-64.5 144-144S383.5 0 304 0 160 64.5 160 144c0 28.2 8.1 54.5 22.1 76.7L9.4 393.4c-12.5 12.5-12.5 32.8 0 45.3l64 64c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 416 96 397.3l41.4 41.4c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L141.3 352l86.1-86.1c22.2 14 48.5 22.1 76.7 22.1z"
        />
      </svg>
      Your user key
    </a>
    <div
      class="userMenuItem"
    >
      <svg
        class="userIcon"
        fill="none"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#user-a)"
        >
          <path
            d="M24.938 24.012A7.998 7.998 0 0 0 18 20h-4a7.998 7.998 0 0 0-6.938 4.012A11.973 11.973 0 0 0 16 28c3.55 0 6.738-1.544 8.938-3.988ZM0 16a16 16 0 1 1 32 0 16 16 0 0 1-32 0Zm16 1a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9Z"
            fill="#AAA"
          />
        </g>
        <defs>
          <clippath
            id="user-a"
          >
            <path
              d="M0 0h32v32H0z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
      <svg
        class="arrowIcon"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#adsa)"
        >
          <path
            d="m23.423 6.718-1.17-1.14a1.86 1.86 0 0 0-1.26-.57 1.8 1.8 0 0 0-1.499.57l-7.498 7.405-7.498-7.405a1.8 1.8 0 0 0-1.5-.57 1.86 1.86 0 0 0-1.259.57L.6 6.718c-.36.335-.575.797-.6 1.289a1.798 1.798 0 0 0 .6 1.379l10.018 10.043a1.8 1.8 0 0 0 1.38.57 1.92 1.92 0 0 0 1.409-.57l10.017-9.953a1.8 1.8 0 0 0 .57-1.47 1.888 1.888 0 0 0-.57-1.288Z"
          />
        </g>
        <defs>
          <clippath
            id="adsa"
          >
            <path
              d="M0 0h24v24H0z"
            />
          </clippath>
        </defs>
      </svg>
    </div>
  </nav>
</div>
`;
